src/libMessage/Messenger.cpp

```diff
@@ -3326,22 +3325,22 @@ bool Messenger::SetDSPoWSubmission(
   NumberToProtobufByteArray<uint128_t, UINT128_SIZE>(
       gasPrice, *result.mutable_data()->mutable_gasprice());
 
-  if (result.data().IsInitialized()) {
-    bytes tmp(result.data().ByteSize());
-    result.data().SerializeToArray(tmp.data(), tmp.size());
+  if (!result.data().IsInitialized()) {
+    LOG_GENERAL(WARNING, "DSPoWSubmission.Data initialization failed.");
+    return false;
+  }
 
-    Signature signature;
-    if (!MultiSig::GetInstance().SignKey(tmp, submitterKey, signature)) {
-      LOG_GENERAL(WARNING, "Failed to sign PoW.");
-      return false;
-    }
+  bytes tmp(result.data().ByteSize());
+  result.data().SerializeToArray(tmp.data(), tmp.size());
 
-    SerializableToProtobufByteArray(signature, *result.mutable_signature());
-  } else {
-    LOG_GENERAL(WARNING, "DSPoWSubmission.Data initialization failed.");
+  Signature signature;
+  if (!MultiSig::GetInstance().SignKey(tmp, submitterKey, signature)) {
```

Comment:
To look into this

Reply:
@sandipbhoir Why it use MultiSig here? It is not using aggregated pubkey.

Reply:
I didn’t use multisig. As with my commit-   af700ad , Schnorr was used. It must have modified in following pr’s for some reason.

Reply:
This one is okay, please refer to #1097 description for the explanation why we use Multisig